src/app/getInitialProps.js

```diff
@@ -108,14 +113,21 @@ export async function getAppInitialProps(
     ? clientCache.clientSideConfiguration
     : await fetchClientSideConfiguration(hostName);
 
+  if (locale && languages.indexOf(locale) !== 0) {
+    languages = [locale, ...languages];
+  }
+
   const locales = expandedPreferredLocales(languages);
+
   const translations = clientCache.translations
     ? clientCache.translations
     : loadExistingLocalizationByPreference(locales);
 
+  const preferredLocale = translations[0].headers.language;
```

Comment:
0: Maybe we want to give access to this as a return value somehow, unrelated to this PR.